javaslang/src/main/java/javaslang/collection/AbstractsQueue.java

```diff
@@ -0,0 +1,100 @@
+package javaslang.collection;
+
+import javaslang.*;
+import javaslang.control.Option;
+
+import java.util.*;
+
+abstract class AbstractsQueue<T, Q extends AbstractsQueue<T, Q>> implements Kind1<Q, T>, Traversable<T> {
```

Comment:
Good point. Let's remove the empty instance and check for isEmpty. We do it like that in other classes that have a backend/delegate.